scilla/src/backend_collector.rs

```diff
@@ -61,27 +61,26 @@ impl<'a, B: Backend> BackendCollector<'a, B> {
 
     fn update_account_storage(&mut self, address: Address, key: H256, value: H256) {
         // If the account does not exist, check the backend, then create it with empty code and storage
-        match self.account_storage_buffered.entry(address) {
-            std::collections::hash_map::Entry::Vacant(e) => {
-                let account = Account {
-                    nonce: self.backend.basic(address).nonce.as_u64(),
-                    code: self.backend.code(address),
-                    storage_root: None,
-                    is_scilla: true,
-                };
+        if let std::collections::hash_map::Entry::Vacant(e) =
+            self.account_storage_buffered.entry(address)
+        {
+            let account = Account {
+                nonce: self.backend.basic(address).nonce.as_u64(),
+                code: self.backend.code(address),
+                storage_root: None,
+                is_scilla: true,
+            };
+
+            e.insert(Some((account, HashMap::from([(key, value)]))));
+        } else {
+            let entry = self.account_storage_buffered.get_mut(&address).unwrap();
```

Comment:
Why did you change from `match` to `if let`? In my opinion, it was clearer before because you handle both cases explicitly rather than defaulting into the `else` and then having to `.unwrap()` once you know the key is there.

Reply:
Perhaps I misunderstood your original comment - I thought this was what you specified. I will switch it to the match again